framework/include/materials/MaterialPropertyStorage.h

```diff
@@ -119,8 +119,12 @@ class MaterialPropertyStorage
   void shift();
 
   /**
-   * Copy material properties from elem_from to elem_to
-   * Thread safe
+   * Copy material properties from elem_from to elem_to. Thread safe.
+   *
+   * WARNING: This is not capable of copying material data to/from elements on other processors.
+   *          It only works if both elem_to and elem_from are both on the local processor.
+   *          We can't currently check to ensure that they're on processor here because this isn't a ParallelObject.
```

Comment:
This also isn't the right interface for copying to other processors... that interface would need lists of element pairs and properties so that efficient buffers could be built and communicated at once.

This is really a pretty specialized function that's only really used for XFEM.  I kind of don't even like that it's here... :-)

None of that has anything to do with this PR.  You were just moving this comment from the `.C` which is fine... it's good to have it here.